src/NLog.Targets.Syslog/MessageCreation/SdElement.cs

```diff
@@ -15,22 +14,22 @@ internal class SdElement
         private static readonly byte[] RightBracketBytes = { 0x5D };
 
         private readonly SdId sdId;
-        private readonly List<SdParam> sdParams;
+        private readonly IList<SdParam> sdParams;
 
         public SdElement(SdElementConfig sdElementConfig, EnforcementConfig enforcementConfig)
         {
             sdId = new SdId(sdElementConfig.SdId, enforcementConfig);
             sdParams = sdElementConfig.SdParams.Select(sdParamConfig => new SdParam(sdParamConfig, enforcementConfig)).ToList();
         }
 
-        public static void Append(ByteArray message, List<SdElement> sdElements, LogEventInfo logEvent)
+        public static void Append(ByteArray message, IList<SdElement> sdElements, LogEventInfo logEvent)
         {
-            foreach (var sdItem in sdElements)
+            foreach (var sdElement in sdElements)
```

Comment:
This foreach will now allocate with `IList`. `List<SdElement>` will not allocate.